sources/GGTrackerManager.m

```diff
@@ -972,46 +972,61 @@ - (void)startWatchingOrderWithUUID:(NSString *_Nonnull)uuid
             else{
                 
                 // check for share_uuid
-                if (socketResponse && [socketResponse isKindOfClass:[NSDictionary class]]) {
+                if ([socketResponse isKindOfClass:[NSDictionary class]]) {
                     
-                    NSString *shareUUID = shareduuid;
-                    
-                    if (!shareUUID) {
-                        shareUUID = [socketResponse objectForKey:@"share_uuid"];
+                    BOOL isShareUUIDExpired = NO;
+                    id expiredObj = [socketResponse objectForKey:@"expired"];
+                    if ([expiredObj isKindOfClass:[NSNumber class]]) {
+                        isShareUUIDExpired = ((NSNumber *)expiredObj).boolValue;
                     }
-
-                    GGSharedLocation *sharedLocation  = [[GGSharedLocation alloc] initWithData:[socketResponse objectForKey:@"shared_location"] ];
-                    
-                    // updated the order model
-                    activeOrder.sharedLocationUUID = shareUUID;
-                    activeOrder.sharedLocation = sharedLocation;
-                    [_liveMonitor addAndUpdateOrder:activeOrder];
                     
-                    if (self.httpManager && shareUUID) {
-                        // try to get the full order object once
-                        [self startRESTWatchingOrderByOrderUUID:uuid sharedUUID:shareUUID withCompletionHandler:^(BOOL success, NSDictionary * _Nullable response, GGOrder * _Nullable order, NSError * _Nullable error) {
-                           
-                            if (success && order) {
-                                order.sharedLocation = sharedLocation;
-                                
-                                [_liveMonitor addAndUpdateOrder:order];
+                    if (isShareUUIDExpired) {
+                        [activeOrder setStatus:OrderStatusDone];
+                        
+                        if ([delegateOfOrder respondsToSelector:@selector(watchOrderSucceedForOrder:)]) {
+                            [delegateOfOrder watchOrderSucceedForOrder:activeOrder];
+                        }
```

Comment:
test this new flow (success = true, expired = true)
also test what happens if (success = false, expired = true)